core/src/com/unciv/ui/pickerscreens/ImprovementPickerScreen.kt

```diff
@@ -50,11 +56,13 @@ class ImprovementPickerScreen(tileInfo: TileInfo, onAccept: ()->Unit) : PickerSc
 
             group.add(image).size(30f).pad(10f)
 
-            var labelText = improvement.name.tr() + " - " + improvement.getTurnsToBuild(currentPlayerCiv) + " {turns}"
+            var labelText = improvement.name.tr()
+            val turnsToBuild = improvement.getTurnsToBuild(currentPlayerCiv)
+            if (turnsToBuild > 0) labelText += " - $turnsToBuild {turns}"
             val provideResource = tileInfo.hasViewableResource(currentPlayerCiv) && tileInfo.getTileResource().improvement == improvement.name
             if (provideResource) labelText += "\n"+"Provides [${tileInfo.resource}]".tr()
             val removeImprovement = (improvement.name!=RoadStatus.Road.name
-                    && improvement.name!=RoadStatus.Railroad.name && !improvement.name.startsWith("Remove"))
+                    && improvement.name!=RoadStatus.Railroad.name && !improvement.name.startsWith("Remove") && !improvement.name.startsWith("Cancel"))
```

Comment:
`&& improvement.name == Constants.cancelImprovementOrder`

Reply:
Ooops, did I really miss that one? Must have been 4 in the morning.